ores/lock_manager/poolcounter.py

```diff
@@ -4,31 +4,49 @@
 This is useful to make sure too many connections are not coming from one IP.
 """
 
+import logging
 import socket
 
 from .lock_manager import LockManager
+from ..errors import TimeoutError
+
+logger = logging.getLogger(__name__)
 
 
 class PoolCounter(LockManager):
-    def __init__(self, config):
-        self.server = config['server']
-        self.port = config['port']
+    def __init__(self, primary_node, secondary_node=None):
```

Comment:
We discussed in IRC, but for the record https://www.mediawiki.org/wiki/Extension:PoolCounter#Configuration suggests a different implementation, where we have a cluster of pool counter servers and choose a server based on consistent hashing.  The approach here should still work, however.

Reply:
Yeah both fallback and consistent hashing would work. Consistent hashing is preferable as far as I am concerned, but I am fine with fallback for now.

Reply:
okay, I implement consistent hashing here.